pkg/utils/managedresources/managedresources.go

```diff
@@ -195,10 +197,15 @@ func Delete(ctx context.Context, client client.Client, namespace string, name st
 		Delete(ctx); err != nil {
 		return fmt.Errorf("could not delete managed resource '%s/%s': %w", namespace, name, err)
 	}
-
-	if err := builder.NewSecret(client).
+	// label secret because there is no 1-1 correlation from secret to mr (could be prefixed or not)
+	secret := builder.NewSecret(client).
 		WithNamespacedName(namespace, secretName).
-		Delete(ctx); err != nil {
+		WithLabels(map[string]string{"managed-by": name})
+	// apply the label
+	if err := secret.Reconcile(ctx); err != nil {
+		return err
+	}
```

Comment:
IIUC, your idea is to add a label to the Secret and then wait for the Secret's with that label to be gone in the `WaitUntilDeleted` func.

1. I think it is not the best to have the label added by this utility function. This label should be rather maintained by GRM when it reconciles the MR, Secret. One can use the `WaitUntilDeleted` func without using the `Delete` func. This label right now is part of utility funcs, not part of the controller/API contract.
2. Why we need the label at all? Can't we fetch the secret names from the MR spec?